packages/block-editor/src/hooks/sections.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Filters registered block settings, extending attributes to include `isSection` and `sectionName`.
+ *
+ * @param {Object} settings Original block settings.
+ *
+ * @return {Object} Filtered block settings.
+ */
+export function addAttribute( settings ) {
+	if ( hasBlockSupport( settings, '__experimentalSection', false ) ) {
+		// Allow blocks to specify their own metadata attribute definition with default value if needed.
+		if ( ! has( settings.attributes, [ 'metadata' ] ) ) {
```

Comment:
It would be cool to do a dev note to let people know that this might become a "reserved" attribute. We should also document that "metadata", "style", "settings" and "lock" are all reserved attributes.

Writing this makes me thing that "lock" could actually be a child of "metadata" instead of being its own attribute.

Reply:
Are we still on time to make lock a child of metadata? I think it is also fine to have lock as a standalone attribute.

Reply:
Probably too late unless we migrate later (so nothing to do now)